modules/k3s/src/main/java/org/testcontainers/k3s/K3sContainer.java

```diff
@@ -48,41 +52,57 @@ public K3sContainer(DockerImageName dockerImageName) {
         setWaitStrategy(new LogMessageWaitStrategy().withRegEx(".*Node controller sync successful.*"));
     }
 
-    @SneakyThrows
-    public String getKubeConfigYaml(String networkAlias) {
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+        String rawKubeConfig = copyFileFromContainer("/etc/rancher/k3s/k3s.yaml",
+            is -> IOUtils.toString(is, StandardCharsets.UTF_8)
+        );
+        String serverUrl = "https://" + this.getHost() + ":" + this.getMappedPort(KUBE_SECURE_PORT);
+        kubeConfigYaml = kubeConfigWithServerUrl(rawKubeConfig, serverUrl);
+    }
+
+    /**
+     * Return the kubernetes client configuration to access cluster from host machine.
+     *
+     * @return the kubeConfig yaml.
+     */
+    public String getKubeConfigYaml() {
+        return kubeConfigYaml;
+    }
+
+    /**
+     * Generate a kubernetes client configuration for docker internal network. The kubeConfig can be use by
+     * another docker container running in the same network with k3s container. For host network access use
+     * the {@link #getKubeConfigYaml()} method instead.
```

Comment:
Some slight wording tweaks:
```suggestion
     * Generate a kubernetes client configuration for use on a docker internal network. The kubeConfig can be used by
     * another docker container running in the same network as the k3s container. For access from the host, use
     * the {@link #getKubeConfigYaml()} method instead.
```